src/www/views/servers_view/index.ts

```diff
@@ -14,139 +14,155 @@
         height: 192px;
       }
-    </style>
-    <div class="server-list-container">
-      <template is="dom-if" if="[[shouldShowZeroState]]">
-        <div class="flex-column-container">
-          <div class="flex-column-container">
-            <paper-button noink="" on-tap="_requestPromptAddServer">
-              <server-connection-indicator connection-state="disconnected"></server-connection-indicator>
-              <div class="header">[[localize('server-add')]]</div>
-              <div class="subtle">[[localize('server-add-zero-state-instructions')]]</div>
-            </paper-button>
-          </div>
-          <template is="dom-if" if="[[!useAltAccessMessage]]">
-            <div
-              class="footer subtle"
-              inner-h-t-m-l="[[localize('server-create-your-own-zero-state', 'breakLine', '<br/>', 'openLink', '<a href=https://s3.amazonaws.com/outline-vpn/get-started/index.html#step-1>', 'closeLink', '</a>')]]"
-            ></div>
-          </template>
-          <template is="dom-if" if="[[useAltAccessMessage]]">
-            <div
-              class="footer subtle"
-              inner-h-t-m-l="[[localize('server-create-your-own-zero-state-access', 'breakLine', '<br/>', 'openLink', '<a href=https://s3.amazonaws.com/outline-vpn/get-started/index.html#step-1>', 'openLink2', '<a href=https://www.reddit.com/r/outlinevpn/wiki/index/outline_vpn_access_keys/>', 'closeLink', '</a>')]]"
-            ></div>
-          </template>
-        </div>
-      </template>
-      <user-comms-dialog
-        id="autoConnectDialog"
-        localize="[[localize]]"
-        title-localization-key="auto-connect-dialog-title"
-        detail-localization-key="auto-connect-dialog-detail"
-        fire-event-on-hide="AutoConnectDialogDismissed"
-      ></user-comms-dialog>
-      <template is="dom-if" if="[[!shouldShowZeroState]]">
-        <server-list id="serverList" servers="[[servers]]" localize="[[localize]]"></server-list>
-      </template>
-    </div>
-  `,
+    `,
+  ];
+
+  @property({type: Function}) localize: Localizer = msg => msg;
+  @property({type: Boolean}) shouldShowAccessKeyWikiLink = false;
+  @property({type: Array}) servers: ServerListItem[] = [];
 
-  is: 'servers-view',
+  get shouldShowZeroState() {
+    return this.servers ? this.servers.length === 0 : false;
+  }
 
-  properties: {
-    localize: Function,
-    useAltAccessMessage: Boolean,
-    servers: Array,
-    shouldShowZeroState: {
-      type: Boolean,
-      computed: '_computeShouldShowZeroState(servers)',
-    },
-  },
+  private requestPromptAddServer() {
+    this.dispatchEvent(new CustomEvent('add-server', {bubbles: true, composed: true}));
+  }
 
-  _computeShouldShowZeroState(servers: ServerListItem[]) {
-    return servers ? servers.length === 0 : false;
-  },
+  private get zeroStateContent(): DirectiveResult<typeof UnsafeHTMLDirective> {
+    let msg;
+    if (this.shouldShowAccessKeyWikiLink) {
+      msg = this.localize(
```

Comment:
I had the same instinct, went to go do something about it, and then stopped myself because yeah I think it will hurt readability. It would be nice if `localize()` took a map of key: value replacement strings, instead of requiring the arguments to be in `key, value, key, value,...` order. Anyway, that can be a future refactor.

I'm okay with this, as it's already an improvement on what it was before.